jazz_deployments/index.js

```diff
@@ -0,0 +1,502 @@
+// =========================================================================
+// Copyright © 2017 T-Mobile USA, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+
+/**
+  Nodejs Template Project
+  @author: 
+  @version: 1.0
+**/
+
+const errorHandlerModule = require("./components/error-handler.js"); //Import the error codes module.
+const responseObj = require("./components/response.js"); //Import the response module.
+const configObj = require("./components/config.js"); //Import the environment data.
+const logger = require("./components/logger.js"); //Import the logging module.
+const utils = require("./components/utils.js")(); //Import the utils module.
+const validateUtils = require("./components/validation")(); //Import validation module
+const crud = require("./components/crud")(); //Import the crud module.
+const request = require('request');
+const util = require('util');
+
+module.exports.handler = (event, context, cb) => {
+
+	//Initializations
+	var errorHandler = errorHandlerModule(),
+		config = configObj(event);
+	global.config = config;
+	logger.init(event, context);
+
+	//validate inputs
+	logger.info(event);
+	genericInputValidation(event)
+		.then(() => {
+			var deploymentTableName = config.DEPLOYMENT_TABLE,
```

Comment:
Lines 45 - 185 need to be broken down to their own functions. Perhaps along the lines of HTTP "method" and action.

Reply:
Worked on it